tests/test_control_mode.cpp

```diff
@@ -0,0 +1,87 @@
+// -- BEGIN LICENSE BLOCK ----------------------------------------------
+// Copyright 2023 Universal Robots A/S
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//    * Redistributions of source code must retain the above copyright
+//      notice, this list of conditions and the following disclaimer.
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of the {copyright_holder} nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+// -- END LICENSE BLOCK ------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <ur_client_library/comm/control_mode.h>
+
+using namespace urcl;
+
+TEST(control_mode, is_control_mode_realtime_with_realtime_control_modes)
+{
+  for (unsigned int i = 0; i < comm::ControlModeTypes::REALTIME_CONTROL_MODES.size(); ++i)
+  {
+    EXPECT_TRUE(comm::ControlModeTypes::is_control_mode_realtime(comm::ControlModeTypes::REALTIME_CONTROL_MODES[i]));
+  }
+}
+
+TEST(control_mode, is_control_mode_realtime_with_non_realtime_control_modes)
+{
+  for (unsigned int i = 0; i < comm::ControlModeTypes::NON_REALTIME_CONTROL_MODES.size(); ++i)
+  {
+    EXPECT_FALSE(
+        comm::ControlModeTypes::is_control_mode_realtime(comm::ControlModeTypes::NON_REALTIME_CONTROL_MODES[i]));
+  }
+
+  // control modes that are neither realtime or not realtime, but they should also return false
+  EXPECT_FALSE(comm::ControlModeTypes::is_control_mode_realtime(comm::ControlMode::MODE_STOPPED));
```

Comment:
For testing purposes it might make sense to add the third group, so we can iterate over them, as well. Then, we can additionally check that the size of all three groups summed matches the total number of control modes. This way we ensure that adding a control mode later requires grouping that or otherwise the test fails.

Reply:
That's a good idea, I will add a group and a test.